### PR TITLE
Add recipe for Solr 4

### PR DESCRIPTION
**Pull request by stborchert**
For a (Drupal) project we needed to integrate _Apache Solr_ version 4 in our DDEV setup.
The recipe shows a simple example on how to run _Solr_ version 4 in DDEV.

### CODE REVIEW

**Comment by rfay** [comment]
Thanks! Could you say a bit about how this is different from the canonical https://ddev.readthedocs.io/en/stable/users/extend/additional-services/#apache-solr? 

Also, do you recommend updates to https://www.ddev.com/ddev-local/apache-solr-with-drupal-8-and-search-api-solr/ ?

If your recipe isn't explicit about supporting Drupal, please make it more explicit, as there are a lot of differences between supporting different CMSs.

**Comment by stborchert** [comment]
> Could you say a bit about how this is different from the canonical

Ah, of course. Totally forgot about this (though I copied it initially).
Well, the main difference is the _core.properties_ file used in the PR. Instead of using "entrypoint" to create the core, it is automatically created using the data from _core.properties_. I haven't tested yet, if this is the case for newer versions of Solr, too. 
Additionally the Solr configuration isn't copied on startup but it is directly linked from `.ddev\solr\conf`. I'm unsure if there are any downsides of this. At least it works in our project ...

> Also, do you recommend updates to https://www.ddev.com/ddev-local/apache-solr-with-drupal-8-and-search-api-solr/ ?

Hm, not yet :) You highlighted the most important parts ("solr host" and name of the core) so I guess its understandable for users without any (or little) experience on how to setup Solr for Drupal.

> If your recipe isn't explicit about supporting Drupal, please make it more explicit, as there are a lot of differences between supporting different CMSs.

Any hints on how to make this more explicit? I could add an example for the _Search API Server_ configuration override:

```php
/**
 * Override Solr connection.
 */
$config['search_api.server.solr_dev'] = [
  'backend_config' => [
    'connector' => 'standard',
    'connector_config' => [
      'host' => 'solr',
      'path' => '/solr',
      'core' => 'dev',
      'port' => '8983',
      'username' => '',
      'password' => '',
    ]
  ]
];
```

**Comment by rfay** [comment]
I guess I'm concerned that having worked through Solr before, this isn't nearly enough information to help somebody be successful. 

For example, what Drupal module are you using it with? Why did it require Solr 4?

File: docker-compose-services/solr4/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+This recipe allows you to configure a [Solr](https://lucene.apache.org/solr/) server for your project using version 4.
+
+The Solr version used in this recipe is 4.10.4 which is not supported officially anymore.
+
+To enable Solr in your project follow these steps:
+
+1. Create a directory named _solr_ in your project's `.ddev` directory.
+1. Copy [docker-compose.solr.yml](docker-compose.solr.yml) into `.ddev\solr`.
+1. Copy the [core.properties](core.properties) into `.ddev\solr` and edit it according to your needs.
+1. Create a new directory `.ddev\solr\data`.
```
Comment (rfay) [inline docker-compose-services/solr4/README.md:12]: Always use forward slashes, which work everywhere.  Backslashes work only on some Windows shells.
```suggestion
1. Create a new directory `.ddev/solr/data`.
```

File: docker-compose-services/solr4/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+This recipe allows you to configure a [Solr](https://lucene.apache.org/solr/) server for your project using version 4.
+
+The Solr version used in this recipe is 4.10.4 which is not supported officially anymore.
+
+To enable Solr in your project follow these steps:
+
+1. Create a directory named _solr_ in your project's `.ddev` directory.
+1. Copy [docker-compose.solr.yml](docker-compose.solr.yml) into `.ddev\solr`.
+1. Copy the [core.properties](core.properties) into `.ddev\solr` and edit it according to your needs.
```
Comment (rfay) [inline docker-compose-services/solr4/README.md:11]:
```suggestion
1. Copy the [core.properties](core.properties) into `.ddev/solr` and edit it according to your needs.
```

File: docker-compose-services/solr4/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+1. Create a new directory `.ddev\solr\data`.
+1. Copy the configuration suitable for Solr 4.x (including `schema.xml` and `solrconfig.xml`) into a new directory named `.ddev\solr\conf`.
```
Comment (rfay) [inline docker-compose-services/solr4/README.md:13]:
```suggestion
1. Copy the configuration suitable for Solr 4.x (including `schema.xml` and `solrconfig.xml`) into a new directory named `.ddev/solr/conf`.
```

File: docker-compose-services/solr4/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Solr 4.x
+
+This recipe allows you to configure a [Solr](https://lucene.apache.org/solr/) server for your project using version 4.
+
+The Solr version used in this recipe is 4.10.4 which is not supported officially anymore.
+
+To enable Solr in your project follow these steps:
+
+1. Create a directory named _solr_ in your project's `.ddev` directory.
+1. Copy [docker-compose.solr.yml](docker-compose.solr.yml) into `.ddev\solr`.
```
Comment (rfay) [inline docker-compose-services/solr4/README.md:10]:
```suggestion
1. Copy [docker-compose.solr.yml](docker-compose.solr.yml) into `.ddev/solr`.
```

**Review by rfay — CHANGES_REQUESTED** [review]
Various nits above that you can just commit, but the biggest problem is

> I guess I'm concerned that having worked through Solr before, this isn't nearly enough information to help somebody be successful.

> For example, what Drupal module are you using it with? Why did it require Solr 4?

**Comment by rfay** [comment]
Looking forward to seeing this get in @stborchert - Shouldn't take but a couple of minutes...

**Comment by agentrickard** [comment]
I was beating my head against a wall trying to get the 4.x image running using older recipes. This one fixed it right up and let me create a custom core by editing the properties file.

Documentation is clear and clean. Big thumbs up.

**Comment by agentrickard** [comment]
Oddly, however, Search API still says the core is unreachable.

**Review by rfay — APPROVED** [review]
Outstanding, thanks for this. I added an entry in the top-level README and rebased

**Comment by agentrickard** [comment]
A little late the the update party, but I think these changes would affect the solr-5 contrib as well, which I tried to modify without success.

**Pull request merged**